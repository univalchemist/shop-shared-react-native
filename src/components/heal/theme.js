const colors = {
  gray: ['#9e9e9e', '#ececec', '#f1f1f1', '#212121', '#666666', '#f9f9f9'],
  ming: '#39827e',
  teal: '#00847F',
  defaultText: '#333333',
  crimson: '#ED1C24',
  white: '#ffffff',
  black: '#000',
};

colors.backgroundColor = colors.gray[5];

const healTheme = {
  colors,
};
export default healTheme;
