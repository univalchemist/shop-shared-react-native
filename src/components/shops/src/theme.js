const colors = {
  primary: ['#DB0011', '#212121'],
  text: '#212121',
  background: '#F9F9F9',
  border: '#9E9E9E',
  label: '#666666',
  divider: 'rgba(0, 0, 0, 0.05)',
  white: '#fff',
  successGreen: '#00847F',
  disabled: '#D7D8D6',
  ratings: '#646464',
  sliderDisabled: '#D1D1D1',
  transparent: 'transparent',
  backgroundSnackError: '#F6E5E2',
  iconSnackError: '#BF7353',
  backgroundSnackSuccess: '#484848',
};

export default { colors };
