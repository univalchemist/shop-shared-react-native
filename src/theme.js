import { Platform } from 'react-native';
import shopTheme from '@shops/theme';
import healTheme from '@heal/theme';

const colors = {
  primary: ['#DB0011', '#AD0012', '#ffcccf'],
  secondary: ['#2f3494', '#648be7', '#cfe1ff'],
  gray: [
    '#212121',
    '#646464',
    '#727272',
    '#9a9a9a',
    '#b8b8b8',
    '#e3e3e3',
    '#f0f0f0',
    '#f9f9f9',
    '#666666',
    '#f2f2f2',
    '#ededed',
  ],
  success: ['#03A450', '#77b03c', '#dff0d8'],
  warning: ['#935c01', '#ffae28', '#fdf1b2', '#F4511E'],
  error: ['#b00020', '#f6710a', '#fcd1b9', '#d7d8d6'],
  info: ['#31707e', '#1d5b6a', '#d9edf7'],
  black: '#000000',
  white: '#ffffff',
  blue: ['#0275d8', '#afdaff', '#3a98c9', '#294B71'],
  transparent: 'transparent',
  fonts: {
    link: '#0275D8',
    blackLink: '#333333',
  },
  amber: ['#FF9900'],
  red: ['#DB0011'],
  green: ['#00847F'],
};

colors.backgroundColor = colors.gray[7];
colors.buttonPrimary = colors.primary[0];
colors.buttonPrimaryPressed = colors.primary[1];
colors.touchableOverlayColor = colors.gray[6];
colors.riskAmber = colors.amber[0];
colors.riskRed = colors.red[0];
colors.riskGreen = colors.green[0];
colors.imageHeaderMatching = colors.gray[9];
colors.defaultText = '#333333';

const backgroundColor = {
  default: colors.gray[7],
};
const inputField = {
  inputText: colors.gray[1],
  inputBorder: colors.black,
  inputBorderUntouched: colors.gray[3],
  inputFocus: colors.blue[0],
  rightText: 'rgba(0, 0, 0, 0.38)',
  disabledBorder: '#d1d1d1',
  disabledText: 'rgba(0, 0, 0, 0.38)',
};
const healthScoreGraphicColors = {
  lowScore: colors.primary[0],
  averageScore: colors.warning[1],
  highScore: colors.success[0],
  default: colors.gray[5],
  none: colors.transparent,
};
const tabbar = {
  activeTextColor: '#DB0011',
  activeColor: '#DB0011',
  inactiveColor: '#bdbdbd',
};
const checkBox = {
  text: colors.gray[0],
  checked: colors.blue[0],
  unchecked: colors.gray[3],
};
const radioButton = {
  checked: '#00847f',
};

const terminatedLabel = {
  backgroundColor: 'rgba(233, 161, 21, 0.1)',
};

const breakpoints = ['420px', '740px', '980px', '1300px'];

// space is used for margin and padding scales
// it's recommended to use powers of two to ensure alignment
// when used in nested elements
// numbers are converted to px
const space = [0, 4, 8, 16, 32, 64, 128];

// typographic scale
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 96, 128];

// for any scale, either array or objects will work
const lineHeights = [1, 1.125, 1.25, 1.5, 20, 37];

const fonts = {
  default: 'Arial',
};

const fontWeights = {
  light: 300,
  normal: 500,
  bold: 700,
  bolder: Platform.OS === 'ios' ? 800 : 'bold',
};

const letterSpacings = {
  normal: 'normal',
  caps: '0.25em',
};

// border-radius
const radii = [0, 2, 4, 8];

const borders = [0, '1px solid', '2px solid'];

const shadows = [
  `0 1px 2px 0 ${colors.gray[3]}`,
  `0 1px 4px 0 ${colors.gray[3]}`,
];

const customInputStyles = {
  customLabelStyles: {
    color: colors.fonts.blackLink,
  },
  customFocusedLabelStyles: {
    color: colors.fonts.blackLink,
  },
  customHintStyles: { color: colors.fonts.blackLink },
};
const modal = {
  text: colors.white,
  errorText: '#F4511E',
  cancelText: '#00847F',
};
const theme = {
  modal,
  breakpoints,
  colors,
  backgroundColor,
  inputField,
  healthScoreGraphicColors,
  checkBox,
  space,
  fontSizes,
  fonts,
  lineHeights,
  fontWeights,
  letterSpacings,
  radii,
  borders,
  shadows,
  radioButton,
  tabbar,
  terminatedLabel,
  customInputStyles,
  shop: shopTheme,
  heal: healTheme,
};

export default theme;
