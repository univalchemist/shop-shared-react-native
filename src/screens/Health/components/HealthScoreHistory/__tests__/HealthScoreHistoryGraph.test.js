import React from 'react';
import { renderForTest } from '@testUtils';
import HealthScoreHistoryGraph from '../HealthScoreHistoryGraph';
import { BarChart } from 'react-native-svg-charts';
import {
  data,
  dataIncomplete,
  dataWithExcessDataPoints,
  dataConsumedWithExcessDataPoints,
  dataConsumedIncomplete,
  dataConsumed,
} from '../testData';
import theme from '@theme';

const removeOnPressFunction = data => {
  return data.map(point => ({
    value: point.value,
    createdOn: point.createdOn,
  }));
};

describe('HealthScoreHistoryGraph', () => {
  it('should match snapshot', () => {
    const healthScoreHistory = renderForTest(
      <HealthScoreHistoryGraph data={data} />,
    ).toJSON();

    expect(healthScoreHistory).toMatchSnapshot();
  });

  it('should pass data to BarChart canvas', () => {
    const component = renderForTest(<HealthScoreHistoryGraph data={data} />);
    const passedData = removeOnPressFunction(dataConsumed);

    expect(
      removeOnPressFunction(component.queryAllByType(BarChart)[0].props.data),
    ).toMatchObject(passedData);
  });

  describe('rendering incomplete or invalid data', () => {
    it('should render 6 points data when less than 6 points data is passed ', () => {
      const component = renderForTest(
        <HealthScoreHistoryGraph data={dataIncomplete} />,
      );
      const passedData = removeOnPressFunction(dataConsumedIncomplete);

      expect(
        removeOnPressFunction(component.queryAllByType(BarChart)[0].props.data),
      ).toEqual(passedData);
    });

    it('should render last 6 points of data when more than 6 points data is passed ', () => {
      const component = renderForTest(
        <HealthScoreHistoryGraph data={dataWithExcessDataPoints} />,
      );

      const passedData = removeOnPressFunction(
        dataConsumedWithExcessDataPoints,
      );

      expect(
        removeOnPressFunction(component.queryAllByType(BarChart)[0].props.data),
      ).toEqual(passedData);
    });

    it('should render no date when createdOn date passed is invalid', () => {
      const component = renderForTest(
        <HealthScoreHistoryGraph
          data={[{ score: 10, createdOn: 'invalid-date' }]}
        />,
      );
      const dataRendered = [
        { value: 10, createdOn: '' },
        { value: 0, createdOn: '' },
        { value: 0, createdOn: '' },
        { value: 0, createdOn: '' },
        { value: 0, createdOn: '' },
        { value: 0, createdOn: '' },
      ];
      expect(
        removeOnPressFunction(component.queryAllByType(BarChart)[0].props.data),
      ).toEqual(dataRendered);
    });
  });

  it('should render Vertical Labels', () => {
    const component = renderForTest(<HealthScoreHistoryGraph data={data} />);

    expect(
      component.queryAllByProps({ accessibilityLabel: 'Vertical Label 0' })[0]
        .props.children,
    ).toBe(100);
    expect(
      component.queryAllByProps({ accessibilityLabel: 'Vertical Label 1' })[0]
        .props.children,
    ).toBe(75);
    expect(
      component.queryAllByProps({ accessibilityLabel: 'Vertical Label 2' })[0]
        .props.children,
    ).toBe(50);
    expect(
      component.queryAllByProps({ accessibilityLabel: 'Vertical Label 3' })[0]
        .props.children,
    ).toBe(25);
    expect(
      component.queryAllByProps({ accessibilityLabel: 'Vertical Label 4' })[0]
        .props.children,
    ).toBe(0);
  });

  it('should highlight correct latest bar and render latest score label', () => {
    const testData = [
      {
        score: 100,
        createdOn: '2019-01-12T00:00:00',
      },
      {
        score: 25,
        createdOn: '2019-01-31T00:00:00',
      },
      {
        score: 1,
        createdOn: '2019-02-28T00:00:00',
      },
    ];
    const noOfTestData = testData.length;
    const component = renderForTest(
      <HealthScoreHistoryGraph data={testData} />,
    );

    // Bar color
    expect(
      component.queryAllByType(BarChart)[0].props.data[0].svg.fill,
    ).toEqual(theme.colors.gray[3]);
    expect(
      component.queryAllByType(BarChart)[0].props.data[1].svg.fill,
    ).toEqual(theme.colors.gray[3]);
    expect(
      component.queryAllByType(BarChart)[0].props.data[2].svg.fill,
    ).toEqual(theme.colors.black);
    expect(
      component.queryAllByType(BarChart)[0].props.data[3].svg.fill,
    ).toEqual(theme.colors.gray[3]);
    expect(
      component.queryAllByType(BarChart)[0].props.data[4].svg.fill,
    ).toEqual(theme.colors.gray[3]);
    expect(
      component.queryAllByType(BarChart)[0].props.data[5].svg.fill,
    ).toEqual(theme.colors.gray[3]);

    // Rounded Addendum
    expect(
      component.queryAllByType(BarChart)[0].props.children[2].props.barColors,
    ).toEqual([
      theme.colors.gray[3],
      theme.colors.gray[3],
      theme.colors.black,
      theme.colors.gray[3],
      theme.colors.gray[3],
      theme.colors.gray[3],
    ]);

    //Score Label
    expect(
      component.queryAllByType(BarChart)[0].props.children[0].props
        .activeLabelIndex,
    ).toBe(noOfTestData - 1);
  });

  describe('onPressing bars', () => {
    it('should render correct color for bars on click', () => {
      const component = renderForTest(<HealthScoreHistoryGraph data={data} />);

      for (let onPressedIndex = 0; onPressedIndex < 6; onPressedIndex++) {
        component
          .queryAllByType(BarChart)[0]
          .props.data[onPressedIndex].svg.onPress();
        component
          .queryAllByType(BarChart)[0]
          .props.data.forEach((bar, barIndex) => {
            onPressedIndex === barIndex
              ? expect(bar.svg.fill).toEqual(theme.colors.black)
              : expect(bar.svg.fill).toEqual(theme.colors.gray[3]);
          });
      }
    });

    it('should render correct color on rounded addendum on bar pressed', () => {
      const component = renderForTest(<HealthScoreHistoryGraph data={data} />);
      const colorArray = [
        theme.colors.gray[3],
        theme.colors.gray[3],
        theme.colors.gray[3],
        theme.colors.gray[3],
        theme.colors.gray[3],
        theme.colors.gray[3],
      ];

      for (let i = 0; i < 6; i++) {
        component.queryAllByType(BarChart)[0].props.data[i].svg.onPress();
        const newColorArray = colorArray.map((color, index) =>
          index !== i ? color : theme.colors.black,
        );
        expect(
          component.queryAllByType(BarChart)[0].props.children[2].props
            .barColors,
        ).toEqual(newColorArray);
      }
    });

    it('should activate the correct score label', () => {
      const component = renderForTest(<HealthScoreHistoryGraph data={data} />);

      for (let i = 0; i < 6; i++) {
        component.queryAllByType(BarChart)[0].props.data[i].svg.onPress();
        expect(
          component.queryAllByType(BarChart)[0].props.children[0].props
            .activeLabelIndex,
        ).toBe(i);
      }
    });
  });
});
