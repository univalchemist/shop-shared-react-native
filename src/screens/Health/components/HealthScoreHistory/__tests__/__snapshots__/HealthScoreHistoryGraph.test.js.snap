// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthScoreHistoryGraph should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                          ]
                        }
                      >
                        <Text
                          accessibilityLabel="Lifestyle history"
                          accessible={true}
                          color="#212121"
                          fontFamily="default"
                          fontSize={16}
                          lineHeight={22}
                          style={
                            Array [
                              Object {
                                "color": "#212121",
                                "fontFamily": "Arial",
                                "fontSize": 16,
                                "lineHeight": 22,
                                "textAlign": "center",
                              },
                            ]
                          }
                          textAlign="center"
                        >
                          Lifestyle history
                        </Text>
                        <View
                          accessibilityLabel="Lifestyle Score History Graph showing last 6 records."
                          accessible={true}
                          flexDirection="row"
                          justifyContent="center"
                          mt={3}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginTop": 16,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "width": 22,
                                },
                              ]
                            }
                            width={22}
                          >
                            <RNSVGSvgView
                              bbHeight={240}
                              focusable={false}
                              height={240}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "borderWidth": 0,
                                  },
                                  undefined,
                                  Object {
                                    "opacity": 1,
                                  },
                                  null,
                                ]
                              }
                            >
                              <RNSVGGroup
                                fill={
                                  Array [
                                    0,
                                    4278190080,
                                  ]
                                }
                                fillOpacity={1}
                                fillRule={1}
                                font={Object {}}
                                matrix={
                                  Array [
                                    1,
                                    0,
                                    0,
                                    1,
                                    0,
                                    0,
                                  ]
                                }
                                opacity={1}
                                originX={0}
                                originY={0}
                                propList={Array []}
                                rotation={0}
                                scaleX={1}
                                scaleY={1}
                                skewX={0}
                                skewY={0}
                                stroke={null}
                                strokeDasharray={null}
                                strokeDashoffset={null}
                                strokeLinecap={0}
                                strokeLinejoin={0}
                                strokeMiterlimit={4}
                                strokeOpacity={1}
                                strokeWidth={1}
                                vectorEffect={0}
                                x={0}
                                y={0}
                              >
                                <RNSVGText
                                  content={null}
                                  dx={Array []}
                                  dy={Array []}
                                  fill={
                                    Array [
                                      0,
                                      4284900966,
                                    ]
                                  }
                                  fillOpacity={1}
                                  fillRule={1}
                                  font={
                                    Object {
                                      "fontFamily": "Arial",
                                      "fontSize": 11,
                                      "textAnchor": "end",
                                    }
                                  }
                                  matrix={
                                    Array [
                                      1,
                                      0,
                                      0,
                                      1,
                                      0,
                                      0,
                                    ]
                                  }
                                  opacity={1}
                                  originX={0}
                                  originY={0}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                  rotate={Array []}
                                  rotation={0}
                                  scaleX={1}
                                  scaleY={1}
                                  skewX={0}
                                  skewY={0}
                                  stroke={null}
                                  strokeDasharray={null}
                                  strokeDashoffset={null}
                                  strokeLinecap={0}
                                  strokeLinejoin={0}
                                  strokeMiterlimit={4}
                                  strokeOpacity={1}
                                  strokeWidth={1}
                                  vectorEffect={0}
                                  x={
                                    Array [
                                      22,
                                    ]
                                  }
                                  y={
                                    Array [
                                      44,
                                    ]
                                  }
                                >
                                  <RNSVGTSpan
                                    content="100"
                                    dx={Array []}
                                    dy={Array []}
                                    fill={
                                      Array [
                                        0,
                                        4278190080,
                                      ]
                                    }
                                    fillOpacity={1}
                                    fillRule={1}
                                    font={Object {}}
                                    matrix={
                                      Array [
                                        1,
                                        0,
                                        0,
                                        1,
                                        0,
                                        0,
                                      ]
                                    }
                                    opacity={1}
                                    originX={0}
                                    originY={0}
                                    propList={Array []}
                                    rotate={Array []}
                                    rotation={0}
                                    scaleX={1}
                                    scaleY={1}
                                    skewX={0}
                                    skewY={0}
                                    stroke={null}
                                    strokeDasharray={null}
                                    strokeDashoffset={null}
                                    strokeLinecap={0}
                                    strokeLinejoin={0}
                                    strokeMiterlimit={4}
                                    strokeOpacity={1}
                                    strokeWidth={1}
                                    vectorEffect={0}
                                    x={Array []}
                                    y={Array []}
                                  />
                                </RNSVGText>
                                <RNSVGText
                                  content={null}
                                  dx={Array []}
                                  dy={Array []}
                                  fill={
                                    Array [
                                      0,
                                      4284900966,
                                    ]
                                  }
                                  fillOpacity={1}
                                  fillRule={1}
                                  font={
                                    Object {
                                      "fontFamily": "Arial",
                                      "fontSize": 11,
                                      "textAnchor": "end",
                                    }
                                  }
                                  matrix={
                                    Array [
                                      1,
                                      0,
                                      0,
                                      1,
                                      0,
                                      0,
                                    ]
                                  }
                                  opacity={1}
                                  originX={0}
                                  originY={0}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                  rotate={Array []}
                                  rotation={0}
                                  scaleX={1}
                                  scaleY={1}
                                  skewX={0}
                                  skewY={0}
                                  stroke={null}
                                  strokeDasharray={null}
                                  strokeDashoffset={null}
                                  strokeLinecap={0}
                                  strokeLinejoin={0}
                                  strokeMiterlimit={4}
                                  strokeOpacity={1}
                                  strokeWidth={1}
                                  vectorEffect={0}
                                  x={
                                    Array [
                                      22,
                                    ]
                                  }
                                  y={
                                    Array [
                                      86.5,
                                    ]
                                  }
                                >
                                  <RNSVGTSpan
                                    content="75"
                                    dx={Array []}
                                    dy={Array []}
                                    fill={
                                      Array [
                                        0,
                                        4278190080,
                                      ]
                                    }
                                    fillOpacity={1}
                                    fillRule={1}
                                    font={Object {}}
                                    matrix={
                                      Array [
                                        1,
                                        0,
                                        0,
                                        1,
                                        0,
                                        0,
                                      ]
                                    }
                                    opacity={1}
                                    originX={0}
                                    originY={0}
                                    propList={Array []}
                                    rotate={Array []}
                                    rotation={0}
                                    scaleX={1}
                                    scaleY={1}
                                    skewX={0}
                                    skewY={0}
                                    stroke={null}
                                    strokeDasharray={null}
                                    strokeDashoffset={null}
                                    strokeLinecap={0}
                                    strokeLinejoin={0}
                                    strokeMiterlimit={4}
                                    strokeOpacity={1}
                                    strokeWidth={1}
                                    vectorEffect={0}
                                    x={Array []}
                                    y={Array []}
                                  />
                                </RNSVGText>
                                <RNSVGText
                                  content={null}
                                  dx={Array []}
                                  dy={Array []}
                                  fill={
                                    Array [
                                      0,
                                      4284900966,
                                    ]
                                  }
                                  fillOpacity={1}
                                  fillRule={1}
                                  font={
                                    Object {
                                      "fontFamily": "Arial",
                                      "fontSize": 11,
                                      "textAnchor": "end",
                                    }
                                  }
                                  matrix={
                                    Array [
                                      1,
                                      0,
                                      0,
                                      1,
                                      0,
                                      0,
                                    ]
                                  }
                                  opacity={1}
                                  originX={0}
                                  originY={0}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                  rotate={Array []}
                                  rotation={0}
                                  scaleX={1}
                                  scaleY={1}
                                  skewX={0}
                                  skewY={0}
                                  stroke={null}
                                  strokeDasharray={null}
                                  strokeDashoffset={null}
                                  strokeLinecap={0}
                                  strokeLinejoin={0}
                                  strokeMiterlimit={4}
                                  strokeOpacity={1}
                                  strokeWidth={1}
                                  vectorEffect={0}
                                  x={
                                    Array [
                                      22,
                                    ]
                                  }
                                  y={
                                    Array [
                                      129,
                                    ]
                                  }
                                >
                                  <RNSVGTSpan
                                    content="50"
                                    dx={Array []}
                                    dy={Array []}
                                    fill={
                                      Array [
                                        0,
                                        4278190080,
                                      ]
                                    }
                                    fillOpacity={1}
                                    fillRule={1}
                                    font={Object {}}
                                    matrix={
                                      Array [
                                        1,
                                        0,
                                        0,
                                        1,
                                        0,
                                        0,
                                      ]
                                    }
                                    opacity={1}
                                    originX={0}
                                    originY={0}
                                    propList={Array []}
                                    rotate={Array []}
                                    rotation={0}
                                    scaleX={1}
                                    scaleY={1}
                                    skewX={0}
                                    skewY={0}
                                    stroke={null}
                                    strokeDasharray={null}
                                    strokeDashoffset={null}
                                    strokeLinecap={0}
                                    strokeLinejoin={0}
                                    strokeMiterlimit={4}
                                    strokeOpacity={1}
                                    strokeWidth={1}
                                    vectorEffect={0}
                                    x={Array []}
                                    y={Array []}
                                  />
                                </RNSVGText>
                                <RNSVGText
                                  content={null}
                                  dx={Array []}
                                  dy={Array []}
                                  fill={
                                    Array [
                                      0,
                                      4284900966,
                                    ]
                                  }
                                  fillOpacity={1}
                                  fillRule={1}
                                  font={
                                    Object {
                                      "fontFamily": "Arial",
                                      "fontSize": 11,
                                      "textAnchor": "end",
                                    }
                                  }
                                  matrix={
                                    Array [
                                      1,
                                      0,
                                      0,
                                      1,
                                      0,
                                      0,
                                    ]
                                  }
                                  opacity={1}
                                  originX={0}
                                  originY={0}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                  rotate={Array []}
                                  rotation={0}
                                  scaleX={1}
                                  scaleY={1}
                                  skewX={0}
                                  skewY={0}
                                  stroke={null}
                                  strokeDasharray={null}
                                  strokeDashoffset={null}
                                  strokeLinecap={0}
                                  strokeLinejoin={0}
                                  strokeMiterlimit={4}
                                  strokeOpacity={1}
                                  strokeWidth={1}
                                  vectorEffect={0}
                                  x={
                                    Array [
                                      22,
                                    ]
                                  }
                                  y={
                                    Array [
                                      171.5,
                                    ]
                                  }
                                >
                                  <RNSVGTSpan
                                    content="25"
                                    dx={Array []}
                                    dy={Array []}
                                    fill={
                                      Array [
                                        0,
                                        4278190080,
                                      ]
                                    }
                                    fillOpacity={1}
                                    fillRule={1}
                                    font={Object {}}
                                    matrix={
                                      Array [
                                        1,
                                        0,
                                        0,
                                        1,
                                        0,
                                        0,
                                      ]
                                    }
                                    opacity={1}
                                    originX={0}
                                    originY={0}
                                    propList={Array []}
                                    rotate={Array []}
                                    rotation={0}
                                    scaleX={1}
                                    scaleY={1}
                                    skewX={0}
                                    skewY={0}
                                    stroke={null}
                                    strokeDasharray={null}
                                    strokeDashoffset={null}
                                    strokeLinecap={0}
                                    strokeLinejoin={0}
                                    strokeMiterlimit={4}
                                    strokeOpacity={1}
                                    strokeWidth={1}
                                    vectorEffect={0}
                                    x={Array []}
                                    y={Array []}
                                  />
                                </RNSVGText>
                                <RNSVGText
                                  content={null}
                                  dx={Array []}
                                  dy={Array []}
                                  fill={
                                    Array [
                                      0,
                                      4284900966,
                                    ]
                                  }
                                  fillOpacity={1}
                                  fillRule={1}
                                  font={
                                    Object {
                                      "fontFamily": "Arial",
                                      "fontSize": 11,
                                      "textAnchor": "end",
                                    }
                                  }
                                  matrix={
                                    Array [
                                      1,
                                      0,
                                      0,
                                      1,
                                      0,
                                      0,
                                    ]
                                  }
                                  opacity={1}
                                  originX={0}
                                  originY={0}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                  rotate={Array []}
                                  rotation={0}
                                  scaleX={1}
                                  scaleY={1}
                                  skewX={0}
                                  skewY={0}
                                  stroke={null}
                                  strokeDasharray={null}
                                  strokeDashoffset={null}
                                  strokeLinecap={0}
                                  strokeLinejoin={0}
                                  strokeMiterlimit={4}
                                  strokeOpacity={1}
                                  strokeWidth={1}
                                  vectorEffect={0}
                                  x={
                                    Array [
                                      22,
                                    ]
                                  }
                                  y={
                                    Array [
                                      214,
                                    ]
                                  }
                                >
                                  <RNSVGTSpan
                                    content="0"
                                    dx={Array []}
                                    dy={Array []}
                                    fill={
                                      Array [
                                        0,
                                        4278190080,
                                      ]
                                    }
                                    fillOpacity={1}
                                    fillRule={1}
                                    font={Object {}}
                                    matrix={
                                      Array [
                                        1,
                                        0,
                                        0,
                                        1,
                                        0,
                                        0,
                                      ]
                                    }
                                    opacity={1}
                                    originX={0}
                                    originY={0}
                                    propList={Array []}
                                    rotate={Array []}
                                    rotation={0}
                                    scaleX={1}
                                    scaleY={1}
                                    skewX={0}
                                    skewY={0}
                                    stroke={null}
                                    strokeDasharray={null}
                                    strokeDashoffset={null}
                                    strokeLinecap={0}
                                    strokeLinejoin={0}
                                    strokeMiterlimit={4}
                                    strokeOpacity={1}
                                    strokeWidth={1}
                                    vectorEffect={0}
                                    x={Array []}
                                    y={Array []}
                                  />
                                </RNSVGText>
                              </RNSVGGroup>
                            </RNSVGSvgView>
                          </View>
                          <View
                            flexGrow={1}
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 240,
                                  "width": "100%",
                                }
                              }
                            >
                              <View
                                onLayout={[Function]}
                                style={
                                  Object {
                                    "flex": 1,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
